# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/warp/inlined-accessor-exc-bailout.js
# Commit: 12d4b2481567
# Full Hash: 12d4b24815672b4b54ceb54e422a7d7d1d64542e
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2022-03-26 21:33:56
# Description:
#   Bug 1761492 - Override ResumeMode for exception bailouts. r=iain
#   
#   When bailing out to catch an exception, the ResumeMode from the snapshot doesn't
#   make sense as we're resuming at the start of a catch-block.
#   
# ==============================================================================

diff -r e2b40064eb59 -r 12d4b2481567 js/src/jit-test/tests/warp/inlined-accessor-exc-bailout.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/warp/inlined-accessor-exc-bailout.js	Sat Mar 26 10:25:15 2022 +0000
@@ -0,0 +1,45 @@
+// |jit-test| --fast-warmup
+
+// Tests for exception bailout from inlined getter/setter.
+
+function throwingGetter() {
+    var o = {};
+    var count = 0;
+    Object.defineProperty(o, "getter", {get: function() {
+        if (count++ === 195) {
+            throw 1;
+        }
+    }});
+    var ex = null;
+    try {
+        for (var i = 0; i < 200; i++) {
+            o.getter;
+        }
+    } catch(e) {
+        ex = e;
+    }
+    assertEq(ex, 1);
+    assertEq(count, 196);
+}
+throwingGetter();
+
+function throwingSetter() {
+    var o = {};
+    var count = 0;
+    Object.defineProperty(o, "setter", {set: function(v) {
+        if (count++ === 195) {
+            throw 1;
+        }
+    }});
+    var ex = null;
+    try {
+        for (var i = 0; i < 200; i++) {
+            o.setter = i;
+        }
+    } catch(e) {
+        ex = e;
+    }
+    assertEq(ex, 1);
+    assertEq(count, 196);
+}
+throwingSetter();